src/library.js

```diff
@@ -2152,16 +2152,35 @@ LibraryManager.library = {
     }
 
     var lib = LDSO.loadedLibs[handle];
-#if !WASM_BACKEND
-    symbol = '_' + symbol;
+    var is_main_module = lib.module == Module;
+
+    var mangled = '_' + symbol;
+    var mod_symbol = mangled;
+#if WASM_BACKEND
+    if (!is_main_module) {
+      mod_symbol = symbol;
+    }
+    var asm_symbol = symbol;
+#else
+    var asm_symbol = mangled;
 #endif
-    if (!lib.module.hasOwnProperty(symbol)) {
-      DLFCN.errorMsg = ('Tried to lookup unknown symbol "' + symbol +
+
+    if (!lib.module.hasOwnProperty(mod_symbol)) {
+      DLFCN.errorMsg = ('Tried to lookup unknown symbol "' + mod_symbol +
                              '" in dynamic lib: ' + lib.name);
       return 0;
     }
 
-    var result = lib.module[symbol];
+    var result = lib.module[mod_symbol];
+#if WASM
+    // Attempt to get the real "unwrapped" symbol so we have more chance of
+    // getting wasm function which can be added to a table.
```

Comment:
maybe I don't understand the comment, but there seem to be two issues here - first, that we look on `Module.asm` and not just on `Module`, and second, that this is important for getting a wasm function instead of a JS wrapper (I guess of those that we have in ASSERTIONS builds, async compilation, etc.?) Are the issues separate, or is the last the real issue and the first is how we achieve it?

If so, it might be cleaner to have those functions mark their actual wasm function on them, so
```
var jsWrapperFunction = function() {
  ..call the wasm..
};
jsWrapperFunction.wasmFunction = ..the wasm..
```
Then we could just check for `result.wasmFunction`, and this would not be specific to main modules, and could work for others too. Although, perhaps we don't add those wrappers currently for non-main modules?

Reply:
Its just so that we can get a real wasm function for the table.  I will try your approach.

Reply:
The problem is that the jswrappers are generated statically at compile time.   When they are called they dynamically call the wasm functions.  I tried doing something where the wrappers would replace themselves on first use, but that didn't work for this test where thee function in question was not called before the wasm function was needed.   I suppose I could fixup with wrappers during post wasm instantiate?

Reply:
Do you mind if I do this as a followup?   I have an idea for a bigger change to how `Module['foo']` passthrough wrappers work.  Actually.. i can do it a precursor too.

Reply:
On further investigation this looks like the cleanest way to do this for now.

The wasm functions are not available that time of js wrapper creation so we would have to do something like this:

```
var jsWrapperFunction = function() {
  return Module['asm'][foo'].apply(...)
};
jsWrapperFunction.getWasmFunction() = function() {
  return Module['asm'][foo'];
}
```

This would add another wrapper for every wasm function .. which is a lot more expensive that the 3 lines above.

Reply:
Yeah, the current code may be simplest and best.